src/Model.jl

```diff
@@ -216,4 +226,37 @@ end
 spin_components(model::Model) = spin_components(model.spin_polarization)
 
 
-_is_well_conditioned(A; tol=1e5) = (cond(A) <= tol)
+# prevent broadcast
+import Base.Broadcast.broadcastable
+Base.Broadcast.broadcastable(model::Model) = Ref(model)
+
+
+#=
+There are two types of quantities, depending on how they transform under change of coordinates.
+
+Positions transform with the lattice: r_cart = lattice * r_red. We term them vectors.
+
+Linear forms on vectors (anything that appears in an expression f⋅r)
+transform with the inverse lattice transpose: f_cart = lattice' \ f_red. We term them covectors.
+Examples of covectors are forces.
+
+Reciprocal vectors are a special case: they are covectors, but conventionally have an
+additional factor of 2π in their definition, so they transform rather with 2π times the
+inverse lattice transpose: q_cart = 2π lattice' \ q_red = recip_lattice * q_red.
+=#
+vector_red_to_cart(model::Model, rred) = model.lattice * rred
+vector_cart_to_red(model::Model, rcart) = model.inv_lattice * rcart
```

Comment:
Should add quick tests, just to ensure forward and backward gives an identity.

Reply:
that kind of tests don't actually test anything, because I know they are correct now, we're not likely to break it in future refactoring, and it's tested indirectly by other tests. Also tests are code and as such have a nontrivial maintenance cost that I think it's wise to avoid for this type of things

Reply:
well it's not tested. That's the point. It's not covered.